### DISCUSSION TITLE
[논의] 장바구니 요구사항 수정 논의

### DISCUSSION THREAD

**Discussion by Hyeon-Uk** [woowa-techcamp-2024/Team1-3K1K#60] [General]
### 배경
#54 를 진행하다가 생긴 논의입니다.
장바구니 요구사항중 아래와 같은 항목이 있습니다.

```bash
 재고 수가 없으면 음식 상품을 주문할 수 없고, 해당 상품은 품절 상태로 변경된다.
```

재고 수가 없으면 음식 상품을 주문할 수 없는것까진 장바구니에 담을 수 없지만, 장바구니를 담을 때 재고가 없으면 해당 상품이 품절 상태로 변한다는 뉘앙스가 강한것 같아 아래와 같이 수정하는것이 어떤지 의견 나누고싶습니다.

```
재고 수가 없으면 음식 상품을 주문할 수 없다.
주문 완료 후 재고가 소진되면 메뉴는 품절상태로 변한다.
```
```


### 세부 주제

```bash
1.
2.
3.
```

**Comment by Dr-KoKo** [comment]
장바구니 정보를 클라이언트에 저장하게 되면, 1) 상품이 품절되기 전에 장바구니에 상품을 담을 수 있고, 2)이때 품절 여부는 실제 주문요청이 왔을 때 판단이 가능합니다. 그래서 `선 주문 후 상태갱신`의 형태로 논의가 되었던 것 같아요. 저기서 말하는 "해당 상품은 품절 상태로 변경된다"는 사용자의 화면이 갱신된다는 의미로 작성된 것 같습니다.

위에 말씀하신 바대로 장바구니에 담을 때 재고가 사라지는 것이 아니라 주문이 완료되었을 때 재고가 삭감되는 방식에 대해서는 저 또한 찬성입니다. 카트에 담는다고 재고가 사라지는 것은 실 서비스에서 곤란할 수도 있을 거예요.

**Reply by Hyeon-Uk** [reply]
의견 감사합니다! 다른분들 의견도 모아본 뒤, 결정나면 위키 수정하겠습니다! :)

**Comment by kimhyun5u** [comment]
맞는 결정인 것 같습니다! 해당 방향을 수정해도 좋을 것 같습니다!

**Comment by june-777** [comment]
네네 저도 훨씬 더 명확하게 바뀐 것 같습니다! 유저 스토리도 갱신하면 좋을 것 같아요 :)
